### ISSUE TITLE
argmax error in estimation.py

### ISSUE THREAD

**Issue by HJ234** [broadinstitute/CellBender#207]
Hi, thanks for the great program. 
I'm currently getting an error in some of the cellbender runs while computing denoised counts. 
I'm running the most recent developer version of cellbender 0.3.0 with the options --exclude-feature-types Antibody Capture and --cuda. 
The error seems to be related to the argmax function and only occurs in half of my samples. 
Is this caused by something within the cellranger files and will this be resolved with using only CPU? 

```
cellbender:remove-background: Computing target noise counts per gene for MCKP estimator
cellbender:remove-background: Using MCKP noise targets computed for FPR 0.01
cellbender:remove-background: Computing denoised counts using mckp estimator
cellbender:remove-background: Working on chunk (1/7)
Traceback (most recent call last):
  File "cellbender/bin/cellbender", line 8, in <module>
    sys.exit(main())
  File "cellbender/lib/python3.7/site-packages/cellbender/base_cli.py", line 114, in main
    cli_dict[args.tool].run(args)
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/cli.py", line 185, in run
    return main(args)
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/cli.py", line 230, in main
    posterior = run_remove_background(args)
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/run.py", line 127, in run_remove_background
    file_name=file_name,
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/run.py", line 344, in compute_output_denoised_counts_reports_metrics
    use_multiple_processes=args.use_multiprocessing_estimation,
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/posterior.py", line 273, in compute_denoised_counts
    **kwargs,
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/estimation.py", line 527, in estimate_noise
    verbose=verbose,
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/estimation.py", line 634, in _chunk_estimate_noise
    device='cpu')
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/estimation.py", line 832, in apply_function_dense_chunks
    s = fun(dense_tensor, **kwargs)
  File "cellbender/lib/python3.7/site-packages/cellbender/remove_background/estimation.py", line 157, in torch_argmax
    return x.argmax(dim=-1)
RuntimeError: cannot perform reduction function argmax on a tensor with no elements because the operation does not have an identity
```

Any help would be much appreciated!

**Comment by HJ234** [comment]
Thanks for the super quick reply! Here's the pdf output. 
I used the following options: --expected-cells 8663 --total-droplets-included 30000
[cellbender_bug.pdf](https://github.com/broadinstitute/CellBender/files/11344669/cellbender_bug.pdf)
Do you think there might be an easy fix I could use temporarily?

**Comment by sjfleming** [comment]
Wow this looks like a great sample.  I'm not 100% sure what's causing this, since that tensor where the `argmax()` is being taken should never be empty... but I've made an attempt at a quick fix.

https://github.com/broadinstitute/CellBender/commit/ffce3f201633b940a53ba62334b4c0415c5724b6

Could you try running using this branch
`sf_dev_0.3.0_postreg_posterior_format_h5` and let me know if that fixes it?

**Comment by HJ234** [comment]
It's working!! I just added the two lines of code to estimation.py and it's moving on to the next step (or chunk) now. I'm hoping it works for the rest of the samples. 
Thank you so much for the fix. I really appreciate it!

**Comment by sjfleming** [comment]
Excellent!  I love it when there's a fix that's just two lines of code :)
(Though I still have to dig around and figure out why those two lines were necessary... because I didn't think they were necessary...)

But anyway, hope it works for the other samples too.

**Comment by sjfleming** [comment]
Closed by #238
